library-best-practices/JLBP-6.md

```diff
@@ -2,7 +2,7 @@
 --------------------------------------------
 
 When a library B in Java depends on another library A through the Maven
-repository system, there are two types of handles that library B needs to use to
+repository system, there are two identifiers that library B needs to
```

Comment:
This creates a garden path - "identifiers" fills in the object slot of "access" until "classes" intrudes and bumps it out. Even though it feels more wordy, I think we need either "needs to use to access" or "needs in order to access."

Reply:
"use" is a weak, non-descript verb that's almost never needed, especially in conjunction with any other verb. And "in order to" is simply a wordy synonym for "to".

Reply:
I don't disagree with what you're saying about "use" and "in order to", but the garden path still needs to be fixed somehow.

Reply:
not sure what you mean by "garden path" here

Reply:
https://en.wikipedia.org/wiki/Garden-path_sentence

Read this in isolation:

"there are two identifiers that library B needs to access"

then read:

"there are two identifiers that library B needs to access classes"

The "classes" feels hanging if you read it this way, because you have to re-interpret the grammar (specifically, what "access" applies to).

Reply:
rewritten